### PR TITLE
Feat/api refactor

### PR DESCRIPTION
**Pull request by plutoless**

### CODE REVIEW

File: server/README.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+
+### POST /start
+This api starts an agent with given graph and override properties. The started agent will join into the specified channel, and subscribe to the uid which your browser/device's rtc use to join.
+
+| Param    | Description |
+| -------- | ------- |
+| request_id  | any uuid for tracing purpose    |
+| channel_name | channel name, it needs to be the same with the one your browser/device joins, agent needs to stay with your browser/device in the same channel to communicate  |
+| chat_uid    | the uid which your browser/device's rtc use to join, agent needs to know your rtc uid to subscribe your audio    |
```
Comment (wangyoucao577) [inline server/README.md:18]: Is `chat_uid` a good name here? maybe `uid` is good enough?

**Review by wangyoucao577 — COMMENTED** [review]
What's the main purpose for this refactoring? Any problem to solve?

**Comment by plutoless** [comment]
> What's the main purpose for this refactoring? Any problem to solve?

1. move graph logic to front end service, instead of doing it in front-ui on browser
2. provide core agent mgmt apis and docs, so that api can be used without ui
3. add inifinity timeout option when starting worker, so it's easier for developers who starts agent w/o ui (he does not need to keep pinging)

File: server/README.md (hunk 1 above)
Comment (plutoless) [reply server/README.md:18]: uid is bit ambiguous as it can refer to bot uid.. or maybe we should expose both so that it's clearer?

Comment (plutoless) [reply server/README.md:18]: few options,
1. human uid
2. adding optional bot_uid, so it's clearer when using

Comment (wangyoucao577) [reply server/README.md:18]: Agree that expose both `uid` and `bot_uid`(or maybe `agent_uid`?) seems better.

Comment (plutoless) [reply server/README.md]: end up with user_uid and bot_uid

**Review by wangyoucao577 — APPROVED** [review]

**Pull request merged**